aten/src/ATen/THLongStorageView.h

```diff
@@ -48,38 +41,17 @@ class THLongStorageView {
   */
 
   THLongStorageView(ArrayRef<int64_t> ref, THLongStorageViewKind kind)
-  : zero_dim_to_null(false)
   {
     // zero_dim_to_one converts an empty ArrayRef into [1]
-    // zero_dim_to_null converts an empty ArrayRef into a null THLongStorage
-    // noelem_to_empty makes an ArrayRef of [0] into an empty THLongStorage
-    bool zero_dim_to_one = false;
-    bool noelem_to_empty = false;
-    switch (kind) {
-      case THLongStorageViewKind::SIZE:
-        zero_dim_to_one = true;
-        break;
-      case THLongStorageViewKind::STRIDE_EMPTY_TENSOR:
-        zero_dim_to_null = true;
-        noelem_to_empty = true;
-        break;
-      case THLongStorageViewKind::STRIDE_SCALAR:
-        zero_dim_to_null = true;
-        break;
-      case THLongStorageViewKind::LENGTH:
-        break;
-    }
+    bool zero_dim_to_one = kind == THLongStorageViewKind::SIZE_STRIDE;
+
     if(zero_dim_to_one && ref.size() == 0) {
```

Comment:
And, just to double check for my understanding, this conditional didn't go away because it's handling scalars, and scalars aren't being pushed into TH in this PR.

Reply:
(You did add `USE_TH_SCALAR` though, which puzzles me a little.)

Reply:
I agree it's not super clear.

Your first comment is correct "this conditional didn't go away because it's handling scalars, and scalars aren't being pushed into TH in this PR."

`USE_TH_SCALAR` is there because I basically need to audit all the calls to `_dim()` and the legacy `resize` functions.  Some of them I noticed have code that is incorrect if TH has scalar support.  So it seemed best to fix up everything at once (even if some of the behavior is hidden behind an ifdef) than to leave a FIXME that I will have to figure out again.

Reply:
OK, SGTM.